### PR TITLE
Node overview

### PR DESCRIPTION
**Pull request by rawagner**
screencapture-localhost-9000-k8s-cluster-nodes-ip-10-0-170-157-us-east-2-compute-internal-2020-04-03-16_14_18
screencapture-localhost-9000-k8s-cluster-nodes-ip-10-0-171-115-us-east-2-compute-internal-2020-04-03-16_13_55
Screenshot from 2020-04-03 16-17-23

resource limits/requests (threshold for error was adjusted)
Screenshot from 2020-04-04 16-42-49
Screenshot from 2020-04-04 16-16-33
Screenshot from 2020-04-06 14-27-38

This PR does not contain (will be a separate PR):
 - status messages
 - baremetal info
 - activities

### CODE REVIEW

**Comment by rawagner** [comment]
@andybraren a few notes
Inventory card - Images do not have a link since I think there's no place to link to
Health checks popup - I've changed `Unhealthy conditions` to just `Conditions`

TODO:
 - add resource quotas
 - add activities
 - add messages for status card
 - improve utilization queries

dashboard with baremetal info will be done in separate PR

**Comment by andybraren** [comment]
> Inventory card - Images do not have a link since I think there's no place to link to

I feel like the ability to link to section headings within Details pages has come up before. I don't think we duplicate headings in any individual Details tabs, so using those to generate `id`s seems viable. Maybe implementing this Console-wide would make sense. What do you think @benjaminapetersen?

> Health checks popup - I've changed `Unhealthy conditions` to just `Conditions`

I like that better, good call!

**Comment by kyoto** [comment]
@openshift/openshift-team-monitoring 

Still WIP, but this PR is adding a number of queries (see `queries.ts`).

File: frontend/packages/console-app/src/components/nodes/node-dashboard/queries.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  [NodeQueries.MEMORY_USAGE]: _.template(
+    `node_memory_MemTotal_bytes{instance='<%= node %>'} - node_memory_MemAvailable_bytes{instance='<%= node %>'}`,
+  ),
+  [NodeQueries.MEMORY_TOTAL]: _.template(`node_memory_MemTotal_bytes{instance='<%= node %>'}`),
+  [NodeQueries.POD_COUNT]: _.template(`kubelet_running_pod_count{instance=~'<%= ipAddress %>:.*'}`),
+  [NodeQueries.FILESYSTEM_USAGE]: _.template(
+    `instance:node_filesystem_usage:sum{instance='<%= node %>'}`,
+  ),
+  [NodeQueries.FILESYSTEM_TOTAL]: _.template(
+    `node_filesystem_size_bytes{mountpoint="/", instance='<%= node %>'}`,
```
Comment (paulfantom) [inline frontend/packages/console-app/src/components/nodes/node-dashboard/queries.ts:38]: This is total filesystem size only for `/` mountpoint and not for all node filesystems. Is this intended?

Comment (rawagner) [reply frontend/packages/console-app/src/components/nodes/node-dashboard/queries.ts:38]: no, Im not sure why I added it there.. will be removed

**Comment by brancz** [comment]
Frankly I don't understand why we re-did this work when we already have the Grafana dashboards integrated into the console, and have a node details dashboard as part of that. A number of years have already gone into tuning details of the queries in those dashboards.

**Comment by andybraren** [comment]
@brancz I'll forward you a kinda long email thread from late last year on this topic when those integrated Monitoring Dashboards were being designed.

TL;DR: "Overviews" like this one (this PR is mis-titled) are management-oriented jumping off points with functional and technical differences that distinguish them from the Prometheus-based Cluster/Node Dashboards within the Monitoring area of the Web Console. The Utilization card's queries do overlap, but the various other hyperlinks, statuses, messages, suggested actions, and operator integrations (like metal3) that appear in other cards and popovers address a different set of management-oriented use cases.

FWIW the Console has included node metrics in Details pages for a while, and this new Overview de-emphasizes them somewhat. We'll continue to think about ways to better organize/integrate these views since we're not big fans of overlap either.

image

**Comment by brancz** [comment]
The intend makes perfect sense to me, to me this is just about reusing queries that have been worked on and been tuned for several years as opposed to creating or maintaining the existence of other queries that are likely to repeat the same mistakes. If possible I'd like to see a future where we embed the dashboards that we recently integrated into console for this purpose.

File: frontend/packages/console-app/src/components/nodes/node-dashboard/queries.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
+import * as _ from 'lodash';
+import { QueryWithDescription } from '@console/shared/src/components/dashboard/utilization-card/UtilizationItem';
+
+export enum NodeQueries {
```
Comment (jtomasek) [inline frontend/packages/console-app/src/components/nodes/node-dashboard/queries.ts:4]: Since the Bare Metal Hosts use node exporter queries as well, we should eventually update Bare Metal Host dashboard to use these common queries defined here instead of redefining them specifically for BMHs. (see `metal3-plugin/src/components/baremetal-hosts/dashboard/queries.ts`).
